File: forhksong/A.題目/java/A1-版本轉換/A1-5.java
//====================================
//題目說明：以下程式在 Java 9 SDK可順利編譯，但Java 17 中會產生運行時的警告。
//
//題目要求：請透過LLM及相關tooling，調整程式，使得可以在Java17 環境中運行。
//
//驗證：
//請確認參賽包workshop_tool腳本已運行，其中java-17-app為已準備好的java 17 SDK runtime container，可執行下列指令獲得編譯結果
//
//>> docker run --rm -v $(pwd):/app java-17-app
//>> 會出現Exception in thread "main" java.lang.reflect.InaccessibleObjectException

//內部參考：在 Java 7 中，反射可以輕鬆訪問私有字段，但在 Java 17 的模塊化系統中，這種行為受到了更多限制，會導致 IllegalAccessException。
//====================================
import java.lang.reflect.Field;

public class Main {
    private String secret = "This is a secret";

    public static void main(String[] args) {
        try {
            Main obj = new Main();
            Field field = Main.class.getDeclaredField("secret");
            field.setAccessible(true); // 強制設置可訪問
            String value = (String) field.get(obj);
            System.out.println("Secret value: " + value);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: forhksong/A.題目/python/A1-版本轉換/A1-5.py
##====================================
##題目說明：dataclasses 模組 (Python 3.7+ 支援)
##問題：使用 dataclasses 模組定義資料類型，但該模組在 Python 3.7 之前不存在。
##
##題目要求：請透過LLM及相關tooling，調整程式，使得可以在Java17 環境中運行。
##
##驗證：
##請確認參賽包workshop_tool腳本已運行，其中python-3.x.x-app為已準備好的Python 3.x.x SDK runtime container，可執行下列指令獲得編譯結果
##
##>> docker run --rm -v $(pwd):/app python-3.x.x-app
##====================================

from dataclasses import dataclass

@dataclass
class Person:
    name: str
    age: int

person = Person(name="Alice", age=30)
print(person)  # 錯誤：舊版本不支援 dataclasses 模組



#可能的錯誤訊息： ModuleNotFoundError: No module named 'dataclasses'
#解決方式：在低版本中安裝 dataclasses backport，或手動實現類似的類別結構。